### PR TITLE
Video next button

### PR DESCRIPTION
**Pull request by yichenshen**
Add a next video button.

A slight complication is that we need to use a POST request if the submission does not yet exist. It's done in the same way as the "Watch" button on the index.

### CODE REVIEW

**Comment by yichenshen** [comment]
Yea, I think the submission failure part should be ok since it's pretty much the same thing with the index page. We can see if anyone complains about it first.

I'll fix up the rest, thanks!

**Comment by weiqingtoh** [comment]
> The next video button will fail if the user creates a submission for the next video in another tab after loading the current one. I don't think it is worth the complexity to handle it because the video will likely be open in the browser already, but if you have a simple solution, you can tackle that.

Would it be good if we handle it in the controller action, say if video submission exists, just redirect there?

**Comment by yichenshen** [comment]
@kxmbrian Yup, squashed.

@weiqingtoh I added a redirect on create if a submission exists. Should be ok right?

File: spec/models/course/video_spec.rb [hunk 1]
```diff
@@ -58,6 +58,26 @@
       end
     end
 
+    describe '.next_video' do
```
Comment (kxmbrian) [inline spec/models/course/video_spec.rb]: Sorry to be pedantic, but this should be `#next_video` because it's an instance method.

Comment (yichenshen) [reply spec/models/course/video_spec.rb]: Ic, updated

**Pull request merged**